### ISSUE TITLE
Dev UI chat page: support images in messages

### ISSUE THREAD

**Issue by jmartisk** [quarkiverse/quarkus-langchain4j#458]

**Comment by jmartisk** [comment]
Hm, I'm trying to implement attaching images using the Vaadin upload component and then converting to base64 in the frontend and passing it to a jsonrpc invocation, but I'm hitting some kind of limitation where the websocket for JsonRPC invocations can only handle messages up to 100 kB, anything bigger crashes the websocket :(

@phillip-kruger have you perhaps tried anything similar before?

**Comment by jmartisk** [comment]
I see that another problem will be our usage of `vaadin-message-list`, it doesn't seem to support including images instead of text.

**Comment by phillip-kruger** [comment]
In that case just use normal post, not the ws. You will have to use another comment or build your own

**Comment by jmartisk** [comment]
Yeah that will make things somewhat more complicated because the image will arrive completely independently of the accompanying text message, but I'll give it a try

**Comment by phillip-kruger** [comment]
Shout if I can help

**Comment by jmartisk** [comment]
Hm, I now realize, if I want to add a custom endpoint beyond JSON-RPC  (either a REST endpoint or a reactive route) then I have to add some kind of HTTP extension, and we recently removed that to make quarkus-langchain4j suitable for CLI applications :(

**Comment by phillip-kruger** [comment]
This is only for Dev UI. And you have Verx-http in Dev UI

**Comment by jmartisk** [comment]
We don't have that as a dependency of the extension (neither the deployment, nor the runtime artifact). Quarkus apparently brings it in when the user runs dev mode, but we don't have any compile-time dependency on it (when compiling the extension)

**Comment by phillip-kruger** [comment]
Ok I see. Maybe I can add something to the Dev UI to make this possible.
How urgent is this ?

**Comment by jmartisk** [comment]
It would be cool to have the ability to post images for demoing on conferences (the first is in the half of May), but even if it doesn't work out I still have enough material to present :)

**Comment by jmartisk** [comment]
Simply making the jsonrpc endpoint accept messages over 100 kB isn't possible? I don't know where the limitation is coming from. But lifting it would be the easiest solution.

**Comment by phillip-kruger** [comment]
Yea I am not sure where that comes from. It's not something I set in Dev UI so must be some default of Vertx Web Sockets. If you can find and increase that, go for it !
